Assets/Scripts/PlayerSupervisor.cs

```diff
@@ -219,10 +218,12 @@ void LaunchBall()
 
     public void LoseColliderHit()
     {
+        ball.gameObject.SetActive(false);
+
         if (playerAgent)
             playerAgent.LoseBall();
 
-        LoseGame();
+        gameManager.PlayerLostBall(this);
```

Comment:
Swapping LoseGame for PlayerLostBall here is making players unable to launch returned balls after losing the first one in OnePlayerClearsAllBlocks mode. It seems that the main issue is that without LoseGame being called the PlayerSupervisor never sets its playerState back to Waiting.